pkg/ccl/backupccl/tenant_backup_nemesis_test.go

```diff
@@ -0,0 +1,505 @@
+				impl: func(ctx context.Context, n *randomBackupNemesis, db *gosql.DB) error {
+					if _, err := db.Exec("SET CLUSTER SETTING jobs.debug.pausepoints = 'import.after_ingest'"); err != nil {
+						return err
+					}
+					tableName, err := n.makeRandomBankTable("import_into_cancel")
+					if !testutils.IsError(err, "pause") {
+						return err
+					}
+					if _, err := db.Exec("SET CLUSTER SETTING jobs.debug.pausepoints = ''"); err != nil {
+						return err
+					}
+					for {
+						row := db.QueryRow("SELECT count(1) FROM [SHOW JOBS] WHERE job_type = 'IMPORT' AND status = 'paused'")
+						var count int
+						if err := row.Scan(&count); err != nil {
+							return err
+						}
+						if count > 0 {
+							break
+						}
+						t.Log("waiting for paused import job")
+						time.Sleep(time.Second)
+					}
+					if _, err := db.Exec("CANCEL JOB (SELECT job_id FROM [SHOW JOBS] WHERE job_type = 'IMPORT' AND status = 'paused')"); err != nil {
+						return err
+					}
+					if _, err := db.Exec("SHOW JOBS WHEN COMPLETE (SELECT job_id FROM [SHOW JOBS] WHERE job_type = 'IMPORT')"); err != nil {
+						return err
+					}
+					n.addTable(tableName)
+					return nil
+				},
+			},
+			{name: "IMPORT INTO",
+				impl: func(ctx context.Context, n *randomBackupNemesis, db *gosql.DB) error {
+					tableName, err := n.makeRandomBankTable("import_into")
+					if err != nil {
+						return err
+					}
+					n.addTable(tableName)
+					return nil
+				},
+			},
+			{name: "CREATE INDEX",
+				impl: func(ctx context.Context, n *randomBackupNemesis, db *gosql.DB) error {
+					tableName, err := n.makeRandomBankTable("create_index")
+					if err != nil {
+						return err
+					}
+					if _, err := db.Exec(fmt.Sprintf(`CREATE INDEX ON "%s"(n)`, tableName)); err != nil {
+						return err
+					}
+					n.addTable(tableName)
+					return nil
+				},
+			},
+			{name: "CREATE UNIQUE INDEX (will fail)",
+				impl: func(ctx context.Context, n *randomBackupNemesis, db *gosql.DB) error {
+					tableName, err := n.makeRandomBankTable("create_unique_index")
+					if err != nil {
+						return err
+					}
+					// Add a conflict that ensures the unique index creation will fail.
+					mkConflict := fmt.Sprintf(`UPDATE "%[1]s" SET n = 5 WHERE id IN (SELECT id FROM "%[1]s" ORDER BY random() LIMIT 5)`, tableName)
+					if _, err := db.Exec(mkConflict); err != nil {
+						return err
+					}
+
+					if _, err := db.Exec(fmt.Sprintf(`CREATE UNIQUE INDEX ON "%s"(n)`, tableName)); err == nil {
+						return errors.New("expected error but found none")
+					}
+					n.addTable(tableName)
+					return nil
+				},
+			},
+		},
+	}
+}
+
+func (r *randomBackupNemesis) Start(ctx context.Context) {
+	ctx, cancel := context.WithCancel(ctx)
+
+	r.cancel = cancel
+	r.grp = ctxgroup.WithContext(ctx)
+
+	r.grp.GoCtx(r.runNemesis)
+}
+
+func (r *randomBackupNemesis) Stop() {
+	r.cancel()
+	_ = r.grp.Wait()
+}
+
+func (r *randomBackupNemesis) RequireStart() chan struct{} {
```

Comment:
without comments, I'm having hard time understanding the purpose of all this infra. Could you add a few comments to make it easier for the unfamiliar reader? I believe RequireStart() returns a done channel that the caller should send on when they're ready for the nemesis to begin running, correct?

Reply:
>  I believe RequireStart() returns a done channel that the caller should send on when they're ready for the nemesis to begin running, correct?

Not quite. RequireStart() waits for a new nemesis to start and then returns a channel that will be closed when the nemesis is complete.

I can maybe remove some of this now. The original goal of this was so that you could make sure that a nemesis had started between two incremental backups.  

But, I found during testing, that I actually hit problems more reliably when not enforcing any coordination.

Reply:
If you do prefer leaving the infra in there, perhaps you could add a TODO comment that outlines an example of how it could be used.

Reply:
Its used in the test runner.

Reply:
> Why does the caller need to be aware of the done channel anyway? Why can't RequireStart() close the channel before it returns?

RequireStart returns when the operation _starts_. Done is closed when the operation _completes_. The idea that you can do things like we do at the start of the test runner where you wait for a nemesis to start, start some concurrent operation, and then wait again for it to complete.

> I feel like we should stash any unnecessary infra before merging this PR, and add it back if/when we need it. That way, when this test fails in the wild, it will be easier to debug. If you feel strongly against this, happy to defer to you and keep this stuff in there.

We've now written more lines of commentary than there are lines of code. So I'll just remove it.